### PR TITLE
ContactPreview redesign

### PR DESCRIPTION
**Pull request by murilovmachado**
Jira: https://bugs.benetech.org/browse/CHI-226

This PR implements the visual changes to `ContactPreview` on the Search results.

Screen Shot 2020-09-14 at 17 11 38
Screen Shot 2020-09-14 at 17 12 05

### CODE REVIEW

**Review by nickhurlburt — APPROVED** [review]
@murilovmachado this looks good but I've sent off some product questions, especially on the empty states (like when there's no name).  @GPaoloni please take a look at the code.  Once @GPaoloni has reviewed, I think it's fine if you merge and deploy to staging.  There may be some small tweaks, which could be done in a separate PR.

File: plugin-hrm-form/src/components/search/ContactPreview/ChildNameAndDate.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { PrevNameText, ContactButtonsWrapper, StyledButtonBase, DateText } from '../../../styles/search';
+import { isNonDataCallType } from '../../../states/ValidationRules';
+import CallTypeIcon from '../../common/icons/CallTypeIcon';
+import { channelTypes } from '../../../states/DomainConstants';
+
+const FullscreenIcon = addHover(StyledIcon(Fullscreen));
+
+type OwnProps = {
+  channel: string;
+  callType: string;
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/search/ContactPreview/ChildNameAndDate.tsx:17]: If you address above comment, here is another incidence for those types.

File: plugin-hrm-form/src/components/search/ContactPreview/ConnectContact.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { Template } from '@twilio/flex-ui';
+
+import { isNonDataCallType } from '../../../states/ValidationRules';
+import { StyledButtonBase, ConnectIcon } from '../../../styles/search';
+import { Flex, Box, HiddenText } from '../../../styles/HrmStyles';
+
+type OwnProps = {
+  callType: string;
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/search/ContactPreview/ConnectContact.tsx:10]: And another one.

File: plugin-hrm-form/src/components/search/ContactPreview/TagsAndCounselor.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/search/ContactPreview/TagsAndCounselor.jsx:1]: You don't have to do nothing here, but I'm just curious if adding this file as JS was intentional, and why if so?

**Review by GPaoloni — APPROVED** [review]
Awesome work! Just left some comments on types but no need to worry now if you are with other tasks.

File: plugin-hrm-form/src/components/search/ContactPreview/TagsAndCounselor.jsx (hunk 3 above)
Comment (murilovmachado) [reply plugin-hrm-form/src/components/search/ContactPreview/TagsAndCounselor.jsx:1]: This component was mostly originated by renaming an old one and changing a little bit. But I think it's worth transforming it into a TS file. I'll do that.

**Comment by murilovmachado** [comment]
Deployed to Staging.

**Pull request merged**